chain/cosmos/chain_node.go

```diff
@@ -1291,7 +1292,23 @@ func (tn *ChainNode) CreateNodeContainer(ctx context.Context) error {
 		cmd = []string{chainCfg.Bin, "start", "--home", tn.HomeDir(), "--x-crisis-skip-assert-invariants"}
 	}
 
-	return tn.containerLifecycle.CreateContainer(ctx, tn.TestName, tn.NetworkID, tn.Image, sentryPorts, tn.Bind(), nil, tn.HostName(), cmd, nil)
+	usingPorts := make(nat.PortMap)
+
+	if tn.Index == 0 && chainCfg.HostPortOverride != nil {
+		for intP, extP := range chainCfg.HostPortOverride {
```

Comment:
if you override a single port, do you need to override all ports?

Reply:
no, if a port is not directly overridden a random mapping is assigned. ex:
![image](https://github.com/strangelove-ventures/interchaintest/assets/31943163/7d20798c-cb0f-4026-9d12-ae140cc01649)

Reply:
Ahh - while this logic was correct, it was duplicating bindings to all interfaces and ipv6. So I force the host mappings to ipv4 only now (instead of a blank PortMap it will use the desired ports, then override)

5ea8cd29c2ee6857509e6e7579442cc35b18b2ac